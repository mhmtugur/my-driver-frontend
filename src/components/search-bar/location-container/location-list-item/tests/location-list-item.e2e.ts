import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('location list item', () => {
  let page: E2EPage;
  let element: E2EElement;
  beforeEach(async () => {
    page = await newE2EPage({ html: '<location-list-item></location-list-item>' });
    element = await page.find('location-list-item');
  });
 
  it('should render list item', async () => {
    element.setProperty('location', { label: 'Berlin', address: 'Berlin Airport' });
    await page.waitForChanges();
    let actual = await page.find('.list-item');
    expect(actual).toBeTruthy(); 
    expect(actual.outerHTML).toMatchSnapshot();
  });

  it('should select the location', async () => {
    element.setProperty('location', { label: 'Berlin', address: 'Berlin Airport' });
    const locationSelectedSpy = await page.spyOnEvent('onSelectLocation');
    await page.waitForChanges();
    let listItem = await page.find('.list-item');
    listItem.click();
    await page.waitForChanges();
    expect(locationSelectedSpy.events[0].detail).toEqual({ label: 'Berlin', address: 'Berlin Airport' });
  });
})
