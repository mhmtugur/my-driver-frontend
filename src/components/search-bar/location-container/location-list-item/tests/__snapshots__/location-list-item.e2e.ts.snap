// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location list item should render list item 1`] = `"<div class=\\"list-item\\"><i class=\\"material-icons location-icon\\">location_on</i><div><span class=\\"label\\">Berlin</span><span class=\\"address truncate\\">Berlin Airport</span></div></div>"`;
